src/OAuth/OAuth1/Service/AbstractService.php

```diff
@@ -203,8 +203,11 @@ protected function buildAuthorizationHeaderForAPIRequest(
         }
 
         $parameters = array_merge($parameters, array('oauth_token' => $token->getAccessToken()));
-        $parameters = (is_array($bodyParams)) ? array_merge($parameters, $bodyParams) : $parameters;
-        $parameters['oauth_signature'] = $this->signature->getSignature($uri, $parameters, $method);
+        if(is_array($bodyParams)){
+            $parameters['oauth_signature'] = $this->signature->getSignature($uri, array_merge($parameters, $bodyParams), $method);
+        } else {
+            $parameters['oauth_signature'] = $this->signature->getSignature($uri, $parameters, $method);
+        }
```

Comment:
This is confusing because of duplicate code. A nicer way to write it would be:

```php
if(is_array($bodyParams)) {
    $parameters = array_merge($parameters, $bodyParams);
}
$parameters['oauth_signature'] = $this->signature->getSignature($uri, $parameters, $method);
```